int16 last_player
int16 curr_player
int16 last_move
bool game_end
int16[] board
